client/blb/client.go

```diff
@@ -1127,109 +1126,123 @@ func (cli *Client) readOneTractReplicated(
 	thisB []byte,
 	thisOffset int64) {
 
+	// TODO(eric): remove this var when we redo bad ts version reporting
 	var badVersionHost string
 
 	reqID := core.GenRequestID()
 	order := rand.Perm(len(tract.Hosts))
+	badReadHosts := make(map[string]bool)
 
 	err := core.ErrAllocHost // default error if none present
+	readFunc := func(host string, n int, thisBLoc []byte) tractResult {
+		// TODO(eric): fill in otherHosts when ts-ts cancellation is done.
+		read, err := cli.tractservers.ReadInto(ctx, host, nil, reqID, tract.Tract, tract.Version, thisBLoc, thisOffset)
```

Comment:
I meant instead of using ReadInfo, just use Read. It'll allocate a byte slice only when it needs to (i.e. when a response with data comes back from a server). Allocating them up front might use more memory than it needs to.